src/radroutes.py

```diff
@@ -1,14 +1,92 @@
 #!/usr/bin/env python 
 
 import sys
+import os
+from schema import Schema, SchemaError, And
+import yaml
+import fit2gpx
+import folium
+from folium import plugins
+import pandas as pd
 
 class RadRouter():
-    def process(self, path_to_config):
-        """Consumes a path to a config file and generates a map."""
-        print(f"Path to config file: {path_to_config}")
-        return path_to_config
+    def process_activities(self, path_to_config):
+        # Consumes a path to a config file and generates a map.
+        activities = self.validate_config(path_to_config)
+
+        segments = activities['activities']['segments']
+        directory = activities['activities']['directory']
+
+        conv = fit2gpx.Converter()
+        all_coordinates_list = []
+        map = folium.Map()
+
+        # convert each file to a dataframe and create a list of dataframes
+        for segment in segments:
+            segment_path = directory + segment['file_name']
+            df_laps, df_coordinates = conv.fit_to_dataframes(segment_path)
+            df_coordinates['file_name'] = segment['file_name']
+
+            all_coordinates_list.append(df_coordinates)
+
+        # add check to see if speed exists in dataframe
+        self.add_antpath(all_coordinates_list, map)
+        self.fit_map(all_coordinates_list, map)
+
+        map.save('map.html')
+        print('Map saved.')
+
+    def validate_config(self, path_to_config):
+        try:
+            with open(path_to_config, 'r') as config_file:
+                activities = yaml.safe_load(config_file)
+            print(f"Successfully opened config file: {path_to_config}")
+        except IOError:
+            print(f"Could not find file: {path_to_config}")
+            sys.exit(1)
+
+        # need to add check that individual files exist
+        expected_schema = Schema({
+            'activities': {
+                'directory': os.path.exists,
+                'segments': [{
+                    'file_name': str,
+                    'title': str,
+                    'description': str
+                }]
+            }
+        })
+
+        try:
+            expected_schema.validate(activities)
+            print('Config file is valid.')
+        except SchemaError as se:
+            raise se
+
+        return activities
+
+    def add_antpath(self, coordinates_list, map):
+        # AntPath delay range is 1 to 800. We normalize the max speed from input files against the max of 800.
+        speed = pd.concat(coordinates_list)['speed']
+        normalization = 800 / speed.max()
```

Comment:
To ensure we're not losing any precision (python will truncate by using integer division by default).

```suggestion
        normalization = 800.0 / speed.max()
```